core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -506,6 +514,44 @@ private boolean isThisGraalVMRCObsolete() {
         return false;
     }
 
+    private static String getLinuxID(String option) {
+        Process process = null;
+        BufferedReader reader = null;
+
+        try {
+            StringBuilder responseBuilder = new StringBuilder();
+            String line;
+
+            ProcessBuilder idPB = new ProcessBuilder().command("id", option);
+            idPB.redirectError(new File("/dev/null"));
+
+            try {
+                process = idPB.start();
+                process.getOutputStream().close();
+                reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
```

Comment:
This should just use try-with-resources at each level, as this will ensure that everything is closed.

Reply:
Have resolved with refactor